src/syntax_tree.jl

```diff
@@ -133,7 +133,7 @@ function sourcetext(node::AbstractSyntaxNode)
 end
 
 function Base.range(node::AbstractSyntaxNode)
-    (node.position-1) .+ (1:span(node))
+    node.position:prevind(node.source.code, node.position + span(node))
```

Comment:
It's intentional that `range()` specifically returns a byte range, so this part won't be correct without reviewing the other code.

(In fact, in a change I'm about to do, I'm renaming `range()` to `byte_range()`!)

Reply:
Ah, I see. I think it would be good to have some minimal docs about these things :)

Also, if it is supposed to  _not_ return string indices here, then there is probably some test missing that would actually fail with the changes in my PR here, right?

Reply:
Mm more docs would be good.

At some point this package went from "a hacky thing I dream of being the main parser" to "oh wait it's actually important but still incomplete" :laughing: